Formula/lgeneral.rb

```diff
@@ -0,0 +1,39 @@
+class Lgeneral < Formula
+  desc "Turn-based strategy engine heavily inspired by Panzer General"
+  homepage "https://lgames.sourceforge.io/LGeneral/"
+  url "https://downloads.sourceforge.net/lgeneral/lgeneral/lgeneral-1.4.4.tar.gz"
+  sha256 "0a26b495716cdcab63b49a294ba31649bc0abe74ce0df48276e52f4a6f323a95"
+  license "GPL-2.0-or-later"
+
+  depends_on "gettext"
+  depends_on "sdl12-compat"
+  depends_on "sdl2"
+
+  def install
+    inreplace "configure", "#include <unistd.h>", "#include <sys/stat.h>\n#include <unistd.h>"
+    system "./configure", *std_configure_args,
+                         "--disable-silent-rules",
+                         "--disable-sdltest",
+                         "CFLAGS=-I/opt/homebrew/include/SDL"
+    system "make", "install"
+  end
+
+  def caveats
+    <<~EOS
+      To use add the following into your #{shell_profile}:
+      export DYLD_LIBRARY_PATH="$DYLD_LIBRARY_PATH:$(brew --prefix)/lib"
+
+      Requires pg-data.tar.gz or the original DOS version of Panzer General. Can be downloaded from
+      https://sourceforge.net/projects/lgeneral/files/lgeneral-data/pg-data.tar.gz/download
+      The installation of the data is explained in the file README of pg-data archive.
+
+      mkdir -p #{share}/lgeneral/{nations,scenarios,gfx/{flags,units,terrain},units,sounds,maps}
+      lgc-pg -s <pg-data-unziped-dir> -d #{share}/lgeneral
+    EOS
+  end
+
+  test do
+    puts ENV["DYLD_LIBRARY_PATH"]="/opt/homebrew/lib"
+    system "lgeneral", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This is a GUI program. Unfortunately there is no any cli only functionality which may be treated as a test. Pls see all options below. All of them related to running GUI , except --help. However even `--help` does some positive work, like a cheking library issues, program concitency.  There are some cli tools in this package. I could use them to write test. However it doesn't look like verifing game biniary itself. Anyway, waiting your recomendation. Thnxs.

```
Syntax: lgeneral [options] [resource-file]

Options:
    --ai-debug=<loglev>
		Verbosity level of debug messages for AI turn
    --campaign-start=<state>
		Initial state to start campaign with (debug only)
    --control=<who>
		Entity who controls the current side. <who> is either human,
		or cpu (short: h, or c). Multiple options can be specified.
		In this case the first option refers to the first side, the
		second option to the second side, etc.
    --(no-)deploy-turn
		Allow/Disallow deployment of troops before scenario starts.
    --(no-)purchase
		Enable/Disable option for purchasing by prestige. If disabled
		the predefined reinforcements are used.
    --notitle	Inhibit title screen
    --speed=<n>	Accelerate animation speed by <n>
    --version	Display version information and quit

<resource-file>	scenario file relative to scenarios-directory,
		or campaign file relative to campaigns-directory
		under /opt/homebrew/Cellar/lgeneral/1.4.4/share/lgeneral
```

Reply:
The only option I've found is to gracefully kill the running game process. 
`system "lgeneral & sleep 5; kill $!"`
It works on my laptop, however not sure it seems valuable and  suits for CI.

Reply:
Looks like Linux CI/runner doesn't like to run GUI app without X-Window.
`ERR: sdl_init: No available video deviceLGeneral 1.4.4` 
https://github.com/Homebrew/homebrew-core/runs/6791679231?check_suite_focus=true#step:7:52

Reply:
Will proceed in this thread https://github.com/Homebrew/homebrew-core/pull/102434#discussion_r892500661